File: app/src/main/java/xyz/rodit/snapmod/ObjectProxy.java
package xyz.rodit.snapmod;

import java.lang.reflect.InvocationHandler;
import java.lang.reflect.Method;

public class ObjectProxy implements InvocationHandler {

    private final Object object;

    public ObjectProxy(Object object) {
        this.object = object;
    }

    @Override
    public Object invoke(Object o, Method method, Object[] args) throws Throwable {
        return method.invoke(object, args);
    }
}

File: app/src/main/java/xyz/rodit/snapmod/ObjectProxy.kt
package xyz.rodit.snapmod

import java.lang.reflect.InvocationHandler
import java.lang.reflect.Method

class ObjectProxy(private val target: Any) : InvocationHandler {

    @Throws(Throwable::class)
    override fun invoke(o: Any, method: Method, args: Array<Any>?): Any? {
        if (args == null) return method.invoke(target)
        return method.invoke(target, *args)
    }
}